bayem/checks.py

```diff
@@ -0,0 +1,78 @@
+from collections import defaultdict
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from .vba import DictModelError
+
+
+def _measure(true, lin, norm):
+    a, b = np.asarray(true), np.asarray(lin)
+    error = norm(a - b)
+    ref = norm(b)
+    return error / ref
+
+
+def linearity_analysis(
+    model,
+    posterior,
+    n_sd=3,
+    norm=np.linalg.norm,
+    show=False,
+):
+    """
+    Compares the `model` responses `n_sd` standard deviations around the
+    `posterior` mean with its linearization to estimate a measure of
+    linearity.
+    """
+
+    model = DictModelError(model, jac=None, noise0=None)
+
+    k, J = model.first_call(posterior.mean)
+
+    linearity_measure = defaultdict(dict)
+
+    for i_sd, sd in enumerate(posterior.std_diag):
+        name = posterior.parameter_names[i_sd]
+        single_sd = np.zeros_like(posterior.mean)
+        single_sd[i_sd] = sd
+
+        sd_range = range(-n_sd, n_sd + 1)
```

Comment:
Instead of integers, I guess the input should rather be the interval allowing people to also add floating point numbers, e.g. in the range sd=0..3 with  delta=0.1

Reply:
The user can now directly provide the sd_range. In your example something like
`bayem.check_linearity(..., sd_range=np.linspace(0, 3, 31))`